lib/Serialization/ModuleFile.cpp

```diff
@@ -2959,9 +2959,22 @@ Identifier ModuleFile::getDiscriminatorForPrivateValue(const ValueDecl *D) {
 void ModuleFile::verify() const {
 #ifndef NDEBUG
   const auto &Context = getContext();
-  for (const Serialized<Decl*> &next : Decls)
-    if (next.isComplete() && swift::shouldVerify(next, Context))
+  for (const Serialized<Decl*> &next : Decls) {
+    // Skip any decls that haven't already been deserialized.
+    if (!next.isComplete())
+      continue;
+
+    // Skip verification of generic param decls that haven't yet been parented.
+    // FIXME: This is a hack to workaround the fact that we don't necessarily
+    // parent a GenericTypeParamDecl if we just deserialize its type.
+    if (auto *gpd = dyn_cast<GenericTypeParamDecl>(next.get())) {
+      if (gpd->getDeclContext() == getAssociatedModule())
+        continue;
+    }
+
```

Comment:
Should this be folded into `shouldVerify`?

Reply:
@CodaFi We only want to skip the verification for deserialized generic params, and because the ASTVerifier is defined in libAST, we can't check for `SerializedASTFile` there. We could add an additional flag to the `ModuleDecl`, but given this whole thing is a hack in the first place, I figured it was better to just put it here. What do you think?

Reply:
Oh wait, I guess we could check the `FileUnitKind`...

Reply:
I ended up moving this check into the verification function for `GenericTypeParamDecl`